feat(cli): User Management and Credentials sharing

BHesseldieck opened the pull request.

BHesseldieck committed:
🎉 starting feature development

BHesseldieck committed:
Merge branch 'master' into user-management-p2

BHesseldieck committed:
✨ sharing/unsharing a credential (#3601)

* 🎉 initial design

* ✨ sharing/unsharing of credentials

* ✅ add tests for EE credentials controller

* 💪 implement review comments

* 🛠 refactor agent creation and credential role locking

* 👕 linting adjustments (#3691)

* :shirt: Adjust rule `naming-convention`

* :shirt: Fix `naming-convention` config value

* :shirt: Disregard casing for EE-prefixed vars

Co-authored-by: Iván Ovejero <ivov.src@gmail.com>

ivov committed:
Merge branch 'master' into user-management-p2

BHesseldieck committed:
🛠 refactor authAgents in tests (#3725)

* 🛠 refactor authAgent

* 🛠 refactor authAgent

* 🛠 refactor authAgent

* 🛠 refactor authAgent

* 👕 fix ts issue

* 🐘 add migration for mysql and postgres + add AuthAgent type

Co-authored-by: Iván Ovejero <ivov.src@gmail.com>

valya committed:
⚡️ refactor existing credentials routes (#3672)

* 🎉 initial design

* ✨ sharing/unsharing of credentials

* :recycle: split credential update route into controller and service

* :fire: remove credentials test that is no longer applicable

* :recycle: split credential creation route into controller and service

* :recycle: split single credential get

* :recycle: split delete credentials route

* :recycle: split get all credentials route

* :fire: remove unused imports in credentials contoller

* :fire: remove console.log

* :refactor: changes to credentials controller and service from review

 - removed credentials from service function names
 - made relations list optional
 - put allowGlobalOwner in options objects
 - check length of relations array so join doesn't happen if empty
 - update some comments to further explain rationale
 - remove unneeded `Object.assign`
 - remove non-null assertion from test

* :recycle: move filtered credentials selected fields to variable

* :recycle: remove unneeded merges in credentials service

Co-authored-by: Ben Hesseldieck <b.hesseldieck@gmail.com>
Co-authored-by: Ben Hesseldieck <1849459+BHesseldieck@users.noreply.github.com>

BHesseldieck committed:
✅ fix test

BHesseldieck committed:
🔀 Merge branch 'master' into user-management-p2

BHesseldieck committed:
🐛 fix imports

BHesseldieck committed:
👕 fix lint issue

netroy committed:
User Management: switch over to decorators to define routes (#3827)

BHesseldieck committed:
🔀 Merge branch 'master' into user-management-p2

ivov committed:
Add permissions details to credentials for User Management (#3863)

* :zap: Open `GET /users`

* :zap: Add permissions to cred service

* :truck: Rename method

* :zap: Refactor cred controller

* :test_tube: Adjust test

* :pencil2: Improve comment

* :pencil2: Improve another comment

* :zap: Account for multiple sharings

* :bug: Fix access when user is editor

* :blue_book: Expand interface

* :blue_book: Relocate types

* :blue_book: Exempt cred entity with service-injected fields

* :blue_book: Adjust interface

* :recycle: Add permissions only in `GET /credentials`

* :test_tube: Add expectations for `ownedBy`

* :test_tube: Add sharing details test

* :test_tube: Make `ownedBy` checks more granular

* :blue_book: Adjust interface

* :truck: Rename cred getter

* :recycle: Refactor cred getter

* :test_tube: Expand tests

* :recycle: Refactor to use guard

* :shirt: Remove unneeded lint exception

* :fire: Remove unneeded relation

* :truck: Move relation to `GET /credentials/:id`

* :blue_book: Consolidate typings

* :art: Add multiline for readability

* :fire: Remove unneeded type

* :pencil2: Clarity comment

* :pencil2: Make comments consistent

* :shirt: Add exception to fix build

* :shirt: Add more lint exceptions to fix build

* :bug: Check for non-owner

* :blue_book: Improve typings

* :test_tube: Temporarily skip tests

* :fire: Remove `@ts-ignore`

* :shirt: Move lint exceptions

* :recycle: Refactor cred service and controller

* :zap: Simplify check

BHesseldieck committed:
✏️ adjust naming to experimental

BHesseldieck committed:
⚡️ add credentialsSharing flag to settings

ivov committed:
:twisted_rightwards_arrows: Merge master

BHesseldieck committed:
🛠 add helper to check if UM is also enabled as dependency for CredentialsSharing

BHesseldieck committed:
👕 fix lint error

BHesseldieck committed:
🐘 change name of credential role

BHesseldieck committed:
🚧 WIP batch sharing

BHesseldieck committed:
🚧 WIP use put for sharing

BHesseldieck commented on packages/cli/src/credentials/credentials.service.ee.ts:
```diff
@@ -23,20 +25,45 @@ export class EECredentialsService extends CredentialsService {
+		const role = await RoleService.trxGet(trx, { scope: 'credential', name: 'user' });
+		const users = await UserService.trxGetByIds(trx, shareWith);
+
+		const newSharedCredentials = users
+			.filter((user) => !user.isPending)
+			.map((user) =>
+				Db.collections.SharedCredentials.create({
+					credentials: credential,
+					user,
+					role,
+				}),
+			);
```
I think this would be an edge optimization at the cost of readability

BHesseldieck commented on packages/cli/src/credentials/helpers.ts:
```diff
@@ -1,7 +1,28 @@
+/* eslint-disable no-param-reassign */
```
yes, because we work with the `map` param in the diff

BHesseldieck commented on packages/cli/src/credentials/credentials.controller.ee.ts:
```diff
@@ -21,53 +21,35 @@ EECredentialsController.use((_req, _res, next) => {
 EECredentialsController.post('/:id/share', async (req: CredentialRequest.Share, res) => {
 	const { id } = req.params;
-	const { shareeId } = req.body;
+	const { shareWith } = req.body;
```
@alexgrozav wdyt regarding point 1?

alexgrozav replied:
We can make this change, it's just a one-liner for me.

BHesseldieck replied:
we decided to go with `shareWithIds`

BHesseldieck committed:
✅ add tests for batch sharing, 🛠 implement review suggestions

ivov committed:
✅ expand credential sharing tests for User Management (#3931)

* :test_tube: Expand cred sharing tests

* :zap: Add recently added flags

* ✅ fix and adjust tests for /credentials

Co-authored-by: Ben Hesseldieck <b.hesseldieck@gmail.com>

alexgrozav committed:
:sparkles: User management v2 Front End (#3795)

* feat: Added responsive generic page view layout.

* feat: Added empty state.

* feat: Added credentials view empty state.

* test: Added unit tests for N8nActionBox

* feat: Added credentials list initial design.

* feat: Added credential actions. Started working on filters.

* feat: Updated InfoTip markup, added tests and changed stories to typescript.

* feat: Added credentials filtering by type. Added support for apply/reset filters.

* feat: Added credential sharing user select and user list. Added paywall component.

* feat: Updated credentials view permissions.

* feat: Added support for temporary sharing config for unsaved credentials.

* test: Fixed broken snapshots.

* feat: Added overflow styles to page-view-layout list.

* feat: Handled sharee specific views.

* feat: Integration between FE and BE to support real-world credential sharing scenario.

* feat: Added front end permissions table.

* feat: Refactored credential sharing flow. Updated design elements.

* feat: Added margin and padding auto spacer utilities.

* feat: Rehauled permissions to support instanceOwner role and action inheritance.

* feat: Updated credentials view to apply filters automatically.

* feat: Removed apply filters button and added active button state.

* test: Updated component snapshots.

* refactor: Renamed ResourceSharee to ResourceReader.

* feat: Credential sharing error handling, permissions improvement.

* feat: Updated permissions and error handling.

alexgrozav committed:
chore: Removed console.log.

BHesseldieck committed:
🛠 refactor enabling of credentialsSharing

alexgrozav committed:
feat: Removed owner menu selector from credentials when sharing is disabled.

alexgrozav committed:
Merge branch 'user-management-p2' of github.com:n8n-io/n8n into user-management-p2

alexgrozav committed:
refactor: Moved EE features into ee store module file.

BHesseldieck committed:
🛠 add sharing info to GET credentials/:id

alexgrozav committed:
fix: Fixed initial credential data loading for sharing.

alexgrozav committed:
chore: Removed console.log.

BHesseldieck committed:
🐛 owner can fetch any credential

BHesseldieck committed:
🔀 Merge branch 'master' into user-management-p2

BHesseldieck committed:
🛠 refactor users test

BHesseldieck committed:
👕 fix build type issue

BHesseldieck commented on packages/cli/src/requests.d.ts:
```diff
@@ -106,7 +106,7 @@ export declare namespace CredentialRequest {
 	type Test = AuthenticatedRequest<{}, {}, INodeCredentialTestRequest>;
 
-	type Share = AuthenticatedRequest<{ id: string }, {}, { shareeId: string }>;
+	type Share = AuthenticatedRequest<{ id: string }, {}, { shareWith: string[] }>;
```
we decided to go with `shareWithIds`

alexgrozav committed:
fix: Removed owner tag when credential sharing is disabled. Fixed small reactivity issue.

alexgrozav committed:
chore: Removed console.log.

BHesseldieck committed:
🚧 separate fetching credentials between EE and open

mutdmour commented on packages/editor-ui/src/modules/credentials.ts:
```diff
@@ -150,6 +157,13 @@ const module: Module<ICredentialsState, IRootState> = {
 				return [scopeDefault];
 			};
 		},
+		credentialOwnerName: (state: ICredentialsState, getters: any) =>  // tslint:disable-line:no-any
```
nitpick - `getCredentialOwnerName` is more apt since it returns a function

mutdmour commented on packages/editor-ui/src/modules/credentials.ts:
```diff
@@ -65,8 +68,12 @@ const module: Module<ICredentialsState, IRootState> = {
 		enableOAuthCredential(state: ICredentialsState, credential: ICredentialsResponse) {
 			// enable oauth event to track change between modals
 		},
+		...credentialsEEModule.mutations,
```
maybe we can create a general utility method for extending these modules
at the end of file, can be something like
`export default extendModule(module, credentialsEEModule);`

alexgrozav replied:
Good point! Added a generic typed utility function to extend Vuex modules.

alexgrozav committed:
fix: Fixed empty dropdown in users list.

alexgrozav committed:
fix: Fixed error message and initialization when credential gets unshared.

mutdmour commented on packages/design-system/src/components/N8nUsersList/UsersList.vue:
```diff
@@ -86,19 +101,19 @@ export default mixins(Locale).extend({
 				}
 
-				return a.email > b.email ? 1 : -1;
+				return a.email! > b.email! ? 1 : -1;
```
why is thi sneeded?

alexgrozav replied:
It's not, good point.

mutdmour commented on packages/editor-ui/src/permissions.ts:
```diff
@@ -0,0 +1,71 @@
+ * User permissions definition
+ */
+
+export const getCredentialPermissions = (user: IUser, credential: ICredentialsResponse, store: Store<IRootState>) => {
```
nice.. might help to specify return type

alexgrozav replied:
Done! 🙌

mutdmour commented on packages/editor-ui/src/modules/credentials.ts:
```diff
@@ -170,14 +184,54 @@ const module: Module<ICredentialsState, IRootState> = {
+					credential.ownedBy = data.ownedBy;
+				}
+
+				if (data.sharedWith) {
+					credential.sharedWith = data.sharedWith;
+				}
+
+				context.commit('upsertCredential', credential);
+
+				if (data.sharedWith) {
+					await context.dispatch('setCredentialSharedWith', {
```
same issue as above

alexgrozav replied:
See previous answer.

mutdmour commented on packages/editor-ui/src/components/mixins/nodeHelpers.ts:
```diff
@@ -326,7 +332,11 @@ export const nodeHelpers = mixins(
 						if (nameMatches.length === 0) {
-							foundIssues[credentialTypeDescription.name] = [`Credentials with name "${selectedCredentials.name}" do not exist for "${credentialDisplayName}".`, "You can create credentials with the exact name and then they get auto-selected on refresh."];
+							if (this.$store.getters['settings/isEnterpriseFeatureEnabled'](EnterpriseEditionFeature.Sharing)) {
+								foundIssues[credentialTypeDescription.name] = [`A credential is not accessible`];
+							} else {
+								foundIssues[credentialTypeDescription.name] = [`Credentials with name "${selectedCredentials.name}" do not exist for "${credentialDisplayName}".`, "You can create credentials with the exact name and then they get auto-selected on refresh."];
```
i18n?

alexgrozav replied:
Added i18n for the whole function.

mutdmour commented on packages/editor-ui/src/modules/credentials.ts:
```diff
@@ -170,14 +184,54 @@ const module: Module<ICredentialsState, IRootState> = {
+					credential.ownedBy = data.ownedBy;
+				}
+
+				if (data.sharedWith) {
+					credential.sharedWith = data.sharedWith;
+				}
```
is this correct order? should this happen after `disptch('setCredentialSharedWith'` below instead? in case of error in sharing, would not this be wrong?

alexgrozav replied:
Maybe this could've been improved a bit. I was doing this because the backend wasn't sending me the sharing info in the create and update endpoints.

Updated to be clearer and cleaner. Thanks!

mutdmour commented on packages/editor-ui/src/views/CredentialsView.vue:
```diff
@@ -0,0 +1,430 @@
+		};
+	},
+	computed: {
+		...mapGetters('users', ['currentUser', 'allUsers']),
+		...mapGetters('credentials', ['allCredentials', 'allCredentialTypes', 'credentialTypesById']),
```
nitpick - I have come to hate `mapGetters` because it does not set types.. setting each one alone is annoying but types are helpful

alexgrozav replied:
Migrated to computed functions.

mutdmour commented on packages/editor-ui/src/views/CredentialsView.vue:
```diff
@@ -0,0 +1,430 @@
+	computed: {
+		...mapGetters('users', ['currentUser', 'allUsers']),
+		...mapGetters('credentials', ['allCredentials', 'allCredentialTypes', 'credentialTypesById']),
+		filtersLength() {
```
should set return type so VS code types work

alexgrozav replied:
Done.

mutdmour reviewed (commented):
Love it.. very clean impl.. feel free to treat nitpicks as suggestions and implement what makes sense to you

BHesseldieck committed:
✅ add tests for fetching single credential

netroy committed:
Revert decorators based controllers

alexgrozav commented on packages/design-system/src/components/N8nBadge/Badge.vue:
```diff
@@ -51,10 +51,27 @@ export default Vue.extend({
+	border-radius: var(--border-radius-base);
+	color: var(--color-text-light);
+  	border-color: var(--color-text-light);
+    padding: 1px var(--spacing-4xs);
```
Done.

alexgrozav commented on packages/design-system/src/components/N8nInfoTip/__tests__/InfoTip.spec.ts:
```diff
@@ -0,0 +1,33 @@
+};
+const stubs = ['n8n-tooltip'];
+
+describe('components', () => {
```
Done.

alexgrozav commented on packages/design-system/src/components/N8nInput/__tests__/Input.spec.ts:
```diff
@@ -0,0 +1,11 @@
+import {render} from '@testing-library/vue';
+import N8nInput from "../Input.vue";
+
+describe('components', () => {
```
Done.

alexgrozav commented on packages/design-system/src/components/N8nCard/Card.vue:
```diff
@@ -39,11 +49,27 @@ export default Vue.extend({
 }
+
+.icon {
+  width: 24px;
```
Done.

alexgrozav commented on packages/editor-ui/src/Interface.ts:
```diff
@@ -309,10 +309,16 @@ export interface IActivationError {
 	id: string;
 	createdAt: number | string;
 	updatedAt: number | string;
+	sharedWith: Array<Partial<IUser>>;
+	ownedBy: Partial<IUser>;
```
Good point! Fixed.

alexgrozav commented on packages/editor-ui/src/Interface.ts:
```diff
@@ -726,7 +732,10 @@ export interface IN8nUISettings {
 		path: string;
 	};
+	credentialsSharing: boolean;
+	workflowSharing: boolean;
```
These are no longer used. Good catch!

alexgrozav commented on packages/editor-ui/src/constants.ts:
```diff
@@ -49,6 +49,11 @@ export const COMMUNITY_PACKAGE_MANAGE_ACTIONS = {
 	VIEW_DOCS: 'view-documentation',
 };
 
+export const CREDENTIAL_LIST_ITEM_ACTIONS = {
```
Good point. Moved.

alexgrozav commented on packages/editor-ui/src/components/layouts/PageViewLayout.vue:
```diff
@@ -0,0 +1,79 @@
+<template>
+	<div :class="[$style.wrapper, !sidebarMenuCollapsed && $style['wrapper-with-expanded-sidebar']]">
```
Done.

alexgrozav commented on packages/editor-ui/src/store.ts:
```diff
@@ -121,6 +122,13 @@ export const store = new Vuex.Store({
 	strict: process.env.NODE_ENV !== 'production',
 	modules,
 	state,
+	actions: {
+		async getNodeTypes({ commit, rootGetters }: ActionContext<IWorkflowsState, IRootState>) {
```
Good catch! Refactored.

alexgrozav commented on packages/editor-ui/src/api/nodes.ts:
```diff
@@ -0,0 +1,7 @@
+import {makeRestApiRequest} from "@/api/helpers";
+import {IRestApiContext} from "@/Interface";
+
+export const getNodeTypes = async (context: IRestApiContext, onlyLatest = false): Promise<INodeTypeDescription[]> => {
```
Removed.

alexgrozav commented on packages/editor-ui/src/views/CredentialsView.vue:
```diff
@@ -0,0 +1,430 @@
+			await Promise.all([
+				this.$store.dispatch('credentials/fetchAllCredentials'),
+				this.$store.dispatch('credentials/fetchCredentialTypes'),
+				this.$store.dispatch('getNodeTypes'),
```
Done!

alexgrozav commented on packages/editor-ui/src/components/FeatureComingSoon.vue:
```diff
@@ -1,7 +1,7 @@
 <template>
 	<div v-if="this.featureInfo" :class="$style.container">
 			<div v-if="showHeading" :class="[$style.headingContainer, 'mb-l']">
-				<n8n-heading size="2xlarge">{{ $locale.baseText(featureInfo.featureName) }}</n8n-heading>
+				<n8n-heading size="2xlarge"></n8n-heading>
 			</div>
```
Done.

BHesseldieck committed:
⚡️ adjust credentials test route to also allow testing for sharees (#3999)

* ⚡️ pull data if user is sharee

* fix: Removed sharedWith and ownedBy from credentialData on testing credentials.

Co-authored-by: Alex Grozav <alex@grozav.com>

BHesseldieck committed:
📈 add BE analytics

BHesseldieck committed:
💪 improve credential test

BHesseldieck committed:
⚡️ adjust tracking properties

BHesseldieck committed:
⚡️ removed roles from tracking

BHesseldieck committed:
🔀 Merge branch 'master' into user-management-p2

BHesseldieck committed:
🐛 fix build by removing imports

BHesseldieck committed:
🐛 fix missed merge conflict

alexgrozav committed:
feat: User management P2 Front End bug bash and improvements (#4014)

* fix: Fixed type select size after reopening dropdown.

* fix: Fixed template cards.

* fix: Fixed card content size and copy input.

* fix: Fixed horizontal overflow.

* fix: Hiding el-tags scrollbar in select.

* fix: Added fallback credential icon. Added oAuth credential owner check.

* feat: Added disabled state to user select.

* feat: Added fallback scenario for non-existent credential types.

* feat: Adjusted credentials empty state to show that there are shared credentials.

* fix: Fixed time title.

* feat: Added actionable empty state when shared credentials are present.

* fix: Made action box x padding smaller

* feat: Repositioned owner tag for credential card.

* feat: Updated message box styling to use n8n css variables.

* feat: Added confirmation for deleting sharee.

* fix: Fixed deleted credential types. Fixed select in dropdown bug.

* fix: Various code improvements. Addressed PR review comments.

* fix: Fixed credential deletion errors.

* fix: Various code quality improvements.

BHesseldieck committed:
🔀 Merge branch 'master' into user-management-p2

alexgrozav committed:
feat: N8N-4531 update cloud coming soon features (#4025)

* feat: Showing different upcoming feature messages and format for cloud.

* fix: Changed url format.

* fix: Updated how cloud deployment is determined.

alexgrozav committed:
feat: N8N-4527 implementing credential sharing FE telemetry (#4023)

* feat: Added credential sharing telemetry.

* chore: Renamed computed function for consistency.

* refactor: Simplified subview telemetry sending.

* fix: Changed to callDebounced() helper.

BHesseldieck committed:
🔀 Merge branch 'master' into user-management-p2

BHesseldieck committed:
📧 update email text

alexgrozav committed:
chore: Merge master into user-management-p2

alexgrozav committed:
fix: Adjusted feature coming soon margin.

alexgrozav committed:
chore: Fixed type and line height for delete sharee confirmation modal.

ivov committed:
refactor(editor-ui): Update telemetry (#4040)

* :fire: Remove `identify` from BE

* :zap: Add `versionCli`

* :zap: Add node creator ignore input

* :zap: Move obfuscators to editor-ui

* :zap: Refactor `ph-no-capture`

* :zap: Pass `user_id` to manual exec props

* :truck: Relocate class in `SettingsApiView`

* :zap: Add `userId` to BE PH `identify` call

* :rewind: Revert ":zap: Add `userId` to BE PH `identify` call"

This reverts commit 895aaa45e51506d5dbdcbdabe249a2c743d8e468.

* Revert ":rewind: Revert ":zap: Add `userId` to BE PH `identify` call""

This reverts commit b86a098c202155742c927c88c04c971a5d34dce5.

* :bug: Fix `Promise` handling in `track()` call

* :rewind: Restore `Db.collections` call

* :zap: Set up PH payload to mirror RS

* :fire: Remove excess `userId`

* :blue_book: Remove `userId` from interface

* :fire: Remove unused ref and method

BHesseldieck committed:
🔀 Merge branch 'master' into user-management-p2

alexgrozav committed:
fix: Fixed bug causing instanceOwner to become credential owner on update. (#4079)

BHesseldieck committed:
🐛 fix test for credential shared with member

BHesseldieck committed:
👕 fix lint issues

BHesseldieck committed:
🔀 Merge branch 'master' into user-management-p2

netroy committed:
Merge remote-tracking branch 'origin/master' into user-management-p2

netroy committed:
delete conflicting migration. this data is already seeded in CreateUserManagement

ivov committed:
feat: Expand obfuscation to User Management credential sharing (#4070)

:zap: Expand obfuscation

alexgrozav committed:
feat: Added credential sharing infotip for instance owner.

alexgrozav committed:
Merge branch 'user-management-p2' of github.com:n8n-io/n8n into user-management-p2

netroy committed:
Merge remote-tracking branch 'origin/master' into user-management-p2

netroy committed:
bring back the migration. add a check to avoid conflicts on inserts

krynble commented on packages/cli/src/credentials/credentials.types.ts:
```diff
@@ -0,0 +1,13 @@
+	sharedWith?: UserSharingsDetails[];
+}
+
+export type UserSharingsDetails = {
+	id: string;
+	email: string;
+	firstName: string;
+	lastName: string;
+};
```
Isn't it best if we base it off `User`? Omit some keys or just use parts of it? I'd like to keep a reference to the original object so if for any reason one changes the other is also somewhat reflected.

netroy replied:
created a new interface `IUser` in the workflow package. `User` implements that now.

krynble commented on packages/cli/src/databases/migrations/sqlite/1660062385367-CreateCredentialsEditorRole.ts:
```diff
@@ -0,0 +1,29 @@
+import * as config from '../../../../config';
+import { logMigrationEnd, logMigrationStart } from '../../utils/migrationHelpers';
+
+export class CreateCredentialsEditorRole1660062385367 implements MigrationInterface {
```
Considering the current scope does not allow actually editing the credential when it's shared, should we rename this to `CreateCredentialsUserRole` instead?

krynble commented on packages/workflow/src/Interfaces.ts:
```diff
@@ -124,6 +124,8 @@ export interface ICredentialsDecrypted {
 	type: string;
 	nodesAccess: ICredentialNodeAccess[];
 	data?: ICredentialDataDecryptedObject;
+	ownedBy?: IDataObject;
```
Should we perhaps move the `UserSharingsDetails` type to the `workflow` package and link it here instead of using `IDataObject`? This can then be easily shared with FE for uniformity.

netroy committed:
fix(cli): use a non-env config flag to detect of enterprise features are enabled (#4105)

alexgrozav committed:
chore: Changed ampersand to and in translation.

alexgrozav committed:
Merge branch 'user-management-p2' of github.com:n8n-io/n8n into user-management-p2

ivov committed:
refactor(telemetry): Obfuscate code and JSON editors (#4118)

:zap: Obfuscate code and JSON editors

alexgrozav committed:
feat(editor): improve design and functionality of coming soon features (#4116)

* feat: Improved coming soon feature design and functionality.

* style: Removed empty line.

* chore: Removed unused translation.

netroy committed:
Merge remote-tracking branch 'origin/master' into user-management-p2

netroy committed:
fix: fix telemetry for credential creates and updates (#4125)

fix telemetry for credential creates and updates

netroy committed:
Merge remote-tracking branch 'origin/master' into user-management-p2

krynble committed:
feat: Display errors due to missing credentials in the correct node (#4124)

feat: Display errors due to invalid credentials in the correct node when missing permissions

alexgrozav committed:
fix: remove duplicate header for coming soon features in cloud deployment

netroy committed:
Merge remote-tracking branch 'origin/master' into HEAD

netroy committed:
telemetry: fix the payload for `User viewed credential tab`

netroy committed:
telemetry: add credential_id to 'User selected credential from node modal'

alexgrozav committed:
feat: update empty states for coming soon features

netroy committed:
Update ActionBox.spec.ts.snap

netroy committed:
replace UserSharingsDetails with a subset of User properties

netroy committed:
rename the CreateCredentialsEditorRole to CreateCredentialsUserRole

netroy committed:
move IUser to the workflow package

netroy committed:
use IUser in the frontend as well

netroy committed:
Merge remote-tracking branch 'origin/master' into user-management-p2

janober commented:
Got released with `n8n@0.195.0`
